src/palladio/ShapeGenerator.cpp

```diff
@@ -111,35 +115,104 @@ void ShapeGenerator::get(const GU_Detail* detail, const PrimitiveClassifier& pri
 
 			switch (ar.getStorageClass()) {
 				case GA_STORECLASS_FLOAT: {
-					GA_ROHandleD av(ar);
-					if (av.isValid()) {
-						double v = av.get(primitiveMapOffset);
-						if (DBG)
-							LOG_DBG << "   prim float attr: " << ar->getName() << " = " << v;
-						amb->setFloat(ruleAttrName.c_str(), v);
+					if (isArrayAttribute(ar)) {
+						GA_ROHandleDA av(ar);
+						if (av.isValid()) {
+							UT_Fpreal64Array v;
+							av.get(primitiveMapOffset, v);
+							if (DBG)
+								LOG_DBG << "   prim float array attr: " << ar->getName() << " = " << v;
+							amb->setFloatArray(ruleAttrName.c_str(), v.data(), v.size());
+						}
+					}
+					else {
+						GA_ROHandleD av(ar);
+						if (av.isValid()) {
+							double v = av.get(primitiveMapOffset);
+							if (DBG)
+								LOG_DBG << "   prim float attr: " << ar->getName() << " = " << v;
+							amb->setFloat(ruleAttrName.c_str(), v);
+						}
```

Comment:
Code here (and in other cases) is nested quite deeply. Could maybe be moved into functions to improve readability a bit?

Reply:
Factored out the conversion